woocommerce/woocommerce-gutenberg-products-block
# Reduce the number of dependencies used in Reviews by Product

This PR reduces the number of dependencies used in the _Reviews by Product_ block to this ones:
```diff
-["lodash","wp-api-fetch","wp-compose","wp-date","wp-element","wp-i18n","wp-polyfill","wp-url"]
+["react","react-dom","wp-api-fetch","wp-i18n","wp-polyfill"]
```

All of them are the same as the ones used in the _Product Categories List_ block, but `wp-api-fetch` which is new. I experimented replacing it with `window.fetch()` but `wp-api-fetch` seems to have many things we would need to duplicate in case we didn't use it, so I think it makes sense adding it as a dependency, but happy to hear your thoughts. :slightly_smiling_face: 

In parallel, I completely removed `debounce` without replacing it with anything. Taking a closer look, I thing there was no need to debounce the API calls.

The JS weight required to render a single post with the _Reviews by Product_ block went down from 3.15MB to 1.50MB.

### How to test the changes in this Pull Request:

1. Create a new post and add a _Reviews by Product_ block.
2. Verify everything still works both in the editor and the frontend: try enabling disabling some settings, changing the sort order, loading more reviews, etc.
